qt/scientific_interfaces/Indirect/FqFitDataPresenter.cpp

```diff
@@ -81,55 +35,10 @@ void FqFitDataPresenter::updateActiveDataIndex() { m_dataIndex = m_fqFitModel->g
 
 void FqFitDataPresenter::updateActiveDataIndex(int index) { m_dataIndex = index; }
 
-void FqFitDataPresenter::updateAvailableParameters() { updateAvailableParameters(m_cbParameterType->currentText()); }
-
-void FqFitDataPresenter::updateAvailableParameters(const QString &type) {
-  if (type == "Width")
-    setAvailableParameters(m_fqFitModel->getWidths(TableDatasetIndex{0}));
-  else if (type == "EISF")
-    setAvailableParameters(m_fqFitModel->getEISF(TableDatasetIndex{0}));
-  else
-    setAvailableParameters({});
-
-  if (!type.isEmpty())
-    setSingleModelSpectrum(m_cbParameter->currentIndex());
-}
-
-void FqFitDataPresenter::updateAvailableParameterTypes(FqFitParameters &parameters) {
-  MantidQt::API::SignalBlocker blocker(m_cbParameterType);
-  m_cbParameterType->clear();
-  for (const auto &type : getParameterTypes(parameters))
-    m_cbParameterType->addItem(QString::fromStdString(type));
-}
-
-void FqFitDataPresenter::updateParameterSelectionEnabled() {
-  const auto enabled = m_fqFitModel->getNumberOfWorkspaces() > TableDatasetIndex{0};
-  m_cbParameter->setEnabled(enabled);
-  m_cbParameterType->setEnabled(enabled);
-  m_lbParameter->setEnabled(enabled);
-}
-
-void FqFitDataPresenter::setAvailableParameters(const std::vector<std::string> &parameters) {
-  MantidQt::API::SignalBlocker blocker(m_cbParameter);
-  m_cbParameter->clear();
-  for (const auto &parameter : parameters)
-    m_cbParameter->addItem(QString::fromStdString(parameter));
-}
-
-void FqFitDataPresenter::setParameterLabel(const QString &parameter) { m_lbParameter->setText(parameter + ":"); }
-
-void FqFitDataPresenter::handleParameterTypeChanged(const QString &parameter) {
-  m_lbParameter->setText(parameter + ":");
-  updateAvailableParameters(parameter);
-  emit dataChanged();
-  auto dataType = parameter == QString("Width") ? DataType::WIDTH : DataType::EISF;
-  m_notifier.notify([&dataType](IFQFitObserver &obs) { obs.updateAvailableFunctions(availableFits.at(dataType)); });
-}
-
 void FqFitDataPresenter::setDialogParameterNames(FqFitAddWorkspaceDialog *dialog, const std::string &workspaceName) {
   FqFitParameters parameters;
   try {
-    auto workspace = Mantid::API::AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(workspaceName);
+    auto workspace = ads_instance.retrieveWS<MatrixWorkspace>(workspaceName);
```

Comment:
ads_instance is defined on line 14. if using  Mantid::API::AnalysisDataService::Instance() multilpe time it is usualy better to do it this way rather than initialising it each time